packages/frontend/components/scaffold-eth/AddressInput.tsx

```diff
@@ -0,0 +1,69 @@
+import { ChangeEvent, useCallback, useEffect } from "react";
+import Blockies from "react-blockies";
+import { useEnsAddress } from "wagmi";
+
+interface IAddressInput {
+  value?: string;
+  onChange?: (arg: string) => void;
+  placeholder?: string;
+}
+// can we move this functions to other utility file ?
+const isENS = (address = "") => address.endsWith(".eth") || address.endsWith(".xyz");
+
+export const Sleep = async (time: number): Promise<any> =>
+  new Promise(resolve => setTimeout(() => resolve(true), time));
+
+/**
+ Address input component with ens name resolution
+*/
+
+const AddressInput = ({ value, onChange, placeholder }: IAddressInput): any => {
```

Comment:
I think this components need to have it internal state. We don't want to force the "user" of this component to define an state to use this component.

Reply:
ok moved the states to internal. but should we give the input result with onChange or not? 
using onChange to give ens resolved back let me know its ok or not.

Reply:
Good question!

It's tricky because it'll depend on the use we give to this component. But yeah, I think we should the value back to the parent component.

We have two options here:
1. **onChange**. Send back the raw input of the user (+ the resolved address if he wrote an ENS)
2. **onSuccess**: Send back only when we have a VALID address (because the user pasted the address directly, or because the ENS resolution was succesfulll)

What do you think?

Also, let's not worry too much, we can adapt in the future depending on the needs.